translate/storage/flatxml.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+"""Module for handling flat XML files."""
+
+from lxml import etree
+
+from translate.lang import data
+from translate.misc.xml_helpers import getText, getXMLspace, namespaced
+from translate.storage import base
+from translate.storage.placeables import general
+
+
+class FlatXMLUnit(base.TranslationUnit):
+    """A single term in the XML file."""
+
+    element_name = "str"
+    attribute_name = "key"
+    namespace = None
+    rich_parsers = general.parsers
+    _default_xml_space = "preserve"
+
+    def __init__(self, source=None,
+                 namespace=None, element_name="str", attribute_name="key",
+                 **kwargs):
+        self.namespace = namespace
+        self.element_name = element_name
+        self.attribute_name = attribute_name
+        self.element = etree.Element(self.namespaced(self.element_name))
+        super(FlatXMLUnit, self).__init__(source, **kwargs)
+
+    def __eq__(self, other):
+        """Compares two XML units"""
+        if not isinstance(other, FlatXMLUnit):
+            return super(FlatXMLUnit, self).__eq__(other)
+
+        if self.getid() != other.getid():
+            return False
```

Comment:
I'm not sure if comparing only by id is good idea. On the other side, I have no clue where this is actually being used...

Reply:
It's probably necessary, anyway you might better base your code on LISAunit/LISAfile to get basic xml handling for free...

Reply:
Well, if I remove that comparison, `test_eq` will fail at [`assert unit1 != unit4`](https://github.com/translate/translate/blob/master/translate/storage/test_base.py#L78).

That might be because I don't utilize the passed `source` parameter directly but only hand it to the base class; which writes it to the `source` property which is equal to my id here. I could rewrite it to compare `self.source != other.source` but the effective comparison would be the same.
(same applies to `return self.target == other.target` instead of using `node_text`, because it boils down to the same internally).
And `target` doesn't appear to be set at that location, so the only thing I could compare at this point is `source`/id.

Hence why I added my choice of putting the resource key into `msgid` (which I suppose equals `source`) as discussion point - mostly because I didn't know any better and the resulting PO files looked fine to me.
I'm open for suggestions though if this breaks an assumption anywhere else; or otherwise needs to be handled differently.

Reply:
Just noticed your LISA comment; I deliberately did _not_ base it on LISA because it made a bunch of assumptions and had a few fields that I'd just have to empty out/ignore; so I opted to use the bare metal base class instead.
LISA assumes two levels (`languageNode` and `textNode`) which my format doesn't have (just a single `value_name` element); and I cannot override that from LISA without completely ignoring/rewriting most of the base methods. It is also targetting formats (such as XLIFF) that have base/target languages in them, while my format is monolingual. 
I could rewrite it based on LISA though if you insist (and its entirely possible it does some other stuff I'm not yet aware of; my main use case is to throw it into Weblate).

Reply:
Well your format is quite similar to Android one and that already uses LISA, that's why I was thinking it might be way to go.

Reply:
Hm yeah, I noticed that Android had a very similar structure back when I was exploring options before rolling my own; but it had some quirk that wouldn't work for my files (I think it had to do with quoting or something).
It only uses `LISAfile`, but not `LISAunit`; and even for that there seems to be little it actually uses from it. I'll take another look once I got some free time again; probably won't be able to check on it until the weekend.

Reply:
Well this is really not something what needs to be fixed, I've just mentioned it as an option to get reasonable compliance with the API without too much effort.

Reply:
I don't think there'd be any benefit if I did.
But since I aliased `source` to `getid` and `target` to `node_text` I can actually remove the `__eq__` implementation in favor of `TranslationUnit.__eq__` (which does exactly that). In fact, I can also remove the `getid`/`setid` overloads completely and just use `source`/`target` for even less code overall.